### ISSUE TITLE
Improve and test compatibility with GraphQLJSON

### ISSUE THREAD

**Issue by asbjornenge** [apollostack/apollo-client#217]
Hi! I'm trying to use `apollo-client` with [GraphQLJSON](https://github.com/taion/graphql-type-json).

Using the below `mutation` I am able to send the correct data to GraphQL. The problem arise when I try to query for data that includes `settings` (e.g. the response from this mutation). I get `TypeError: Cannot read property 'selections' of null at writeSelectionSetToStore`. It works if `settings` is null. It also works fine in the `/graphql` query app.

```
function mapMutationsToProps({ ownProps, state }) {
  return {
    updateClub: (variables) => ({
      mutation: gql`
        mutation updateClub(
          $uid: String!
          $name: String!
          $settings: JSON!
        ) {
          updateClub(
            uid: $uid
            name: $name
            settings: $settings
         ) {
            uid,
            name,
            settings
          }
        }
      `,
      variables: variables,
    })
  }
}
```

Any help would be greatly appreciated :smile: :rainbow:

**Comment by asbjornenge** [comment]
I guess the actual question is: "Is there a way to add additional scalar types in apollo-client" ? :stuck_out_tongue:

**Comment by stubailo** [comment]
Can you paste in the response from the graphql server in this case? That will let me write a test to reproduce the issue!

**Comment by asbjornenge** [comment]
This is what I get using `/graphql` (graphiql) if that's what you mean?

```
{
  "data": {
    "updateClub": {
      "uid": "1d7f836018fc11e68d809dfee940f657",
      "name": "Eple",
      "settings": {
        "name": "eple",
        "currency": "AFN",
        "calendarStretch": "2",
        "defaultPreAllocationPeriod": "1",
        "confirmationEmailCopy": null,
        "emailDomains": null
      }
    }
  }
}
```

Or is there some other convenient way to see the reponse?

**Comment by stubailo** [comment]
That's perfect! Another way to see it is to look in your Chrome dev tools under the Network tab.

**Comment by stubailo** [comment]
Got the test going! https://github.com/apollostack/apollo-client/pull/219/commits/4babe248db34474dd914802f379c365b26bbf2d1

**Comment by stubailo** [comment]
Aaaand, it's fixed - that was easy! Now just to wait for CI to pass and republish. https://github.com/apollostack/apollo-client/pull/219

**Comment by stubailo** [comment]
Published in NPM as `0.3.8`, let me know if it works for you!

**Comment by asbjornenge** [comment]
@stubailo works like a charm! Thanks a bunch :heart: :rainbow: :rocket:
